### DISCUSSION TITLE
Can't Figure Out Custom Markers Clustering in Deck.gl

### DISCUSSION THREAD

**Discussion by alitjanjua** [visgl/deck.gl#9220] [Q&A] [answered]
I'm using google maps api to integrate maps to list down US properties with their prices on markers. I want to achieve clustering of pins using my custom svg cluster icon. According to my research it's not possible in deck.gl. When user zoom our, cluster should be created with number of properties in that cluster. My data would be dynamically called via API endpoint so I need the clusters to be dynamically changed and on click of cluster, it should zoom in.
I'm attaching the sample cluster design which I want to achieve with deck.gl. If not possible, I would have to go with google maps clustering.

<img width="921" alt="image" src="https://github.com/user-attachments/assets/c8962f43-bd40-4797-a624-773bdd65ecfb">

**Comment by Pessimistress** [comment, marked as answer]
https://deck.gl/examples/icon-layer

**Reply by alitjanjua** [reply]
It seems to be working on static data. My use cases expects live filtered data while user changes the map bbox. Would it work fine in that case without removing the already loaded pins on map and add all the new pins?

**Reply by Pessimistress** [reply]
I don't see why not.
